pkg/asset/machines/worker.go

```diff
@@ -101,6 +102,10 @@ func defaultBareMetalMachinePoolPlatform() baremetaltypes.MachinePool {
 	return baremetaltypes.MachinePool{}
 }
 
+func defaultVSphereMachinePoolPlatform() vspheretypes.MachinePool {
+	return vspheretypes.MachinePool{}
```

Comment:
Based on reviewing other providers it seems `MachinePool` should contain:
https://github.com/openshift/machine-api-operator/blob/master/pkg/apis/vsphereprovider/v1alpha1/vsphereproviderconfig_types.go#L32-L54

Though I am not sure about `Network NetworkSpec`.

Reply:
Hmm maybe not NetworkSpec since that seems to environmental specific.